### PR TITLE
Update Account Indices section

### LINKED ISSUE #3475: Balance Transfer extrinsic with Index fails - Update Indices doc
Context: https://substrate.stackexchange.com/questions/3620/balance-transfer-fails-when-sending-to-the-accountindex

So, this statement is not accurate https://wiki.polkadot.network/docs/learn-accounts#indices
> Henceforth, you can send me money by just sending to [3] rather than remembering my complete address.

### PR DESCRIPTION
**Pull request by DrW3RK**

### CODE REVIEW

**Comment by DrW3RK** [comment]
When reviewing PRs like these, read them like you are an end user and ensure the instructions make sense by trying them out literally. @alfarok

File: docs/learn/learn-accounts.md [hunk 1]
```diff
@@ -301,35 +301,36 @@ but gets put back when it has the existential deposit.
...
 A Kusama or Polkadot address can have an index. An index is like a short and easy-to-remember
 version of an address. Claiming an index requires a deposit that is released when the index is
-cleared.
+cleared. Any index can be claimed as long as it is not taken by someone else.
 
-Indices are populated in order. Think of them like slots going from 0 to any arbitrary number:
+But what if an account gets reaped as explained above? In that case, the index is emptied. In other
+words, the slot frees up again, making it available for anyone to claim it. It is possible to _freeze_ 
+an index and permanently assign it to an address. This action consumes a deposit but makes sure that 
+the index can never be reclaimed unless released by the holding account.
```
Comment (alfarok) [inline docs/learn/learn-accounts.md:309]: Interesting, didn't know about this yet.

File: docs/learn/learn-accounts.md [hunk 2]
```diff
@@ -301,35 +301,36 @@ but gets put back when it has the existential deposit.
...
-by just sending to `[3]` rather than remembering my complete address.
+When someone shares their account index, their actual account address on-chain can be looked up
+through Polkadot-JS Apps UI > Developer > Chain state > Storage and selecting state query on
+indices pallet for the account corresponding to the index.
 
-```js
-[0][1][2][3][4][5][6]...
-[A][B][C][X][ ][ ][ ]...
-```
+:::
```
Comment (alfarok) [inline docs/learn/learn-accounts.md:317]: I didn't know about this Docusaurus specific markdown syntax either for tips or hints, good to know.

**Review by alfarok — APPROVED** [review]
Tried out the outlined steps locally with no issues.  Looks good to me!

File: docs/learn/learn-accounts.md (hunk 2 above)
Comment (profradha) [reply docs/learn/learn-accounts.md:317]: Yeah, they are called admonitions https://docusaurus.io/docs/markdown-features/admonitions

**Pull request merged**